chainerx_cc/chainerx/cuda/cuda_device.h

```diff
@@ -122,13 +122,13 @@ class CudaDevice : public Device {
 
     // comparison.cu
 
-    void Equal(const Array& x1, const Array& x2, const Array& out) override;
+    void Equal(const Array& x1, const Array& x2, const Array& out, Dtype dtype) override;
```

Comment:
It seems to me that `dtype` is an implementation detail, not something to be specified externally.